Update message deletion info (#123)
xmtp/xmtp-js: README.md

![x-red-sm](https://user-images.githubusercontent.com/510695/163488403-1fb37e86-c673-4b48-954e-8460ae4d4b05.png)

**XMTP client SDK for JavaScript applications**

`xmtp-js` provides a TypeScript implementation of the XMTP client protocol for use with JavaScript and React applications.

> `xmtp-js` is publicly available for building production apps, evaluation, feedback, and community contribution. It has not undergone a formal security audit.

Reviewer: Hmm. I wonder if we should actually aremove this sentence too? There's something that feels tautological about saying "`xmtp-js is publicly available" now that early access is well behind us. And the next sentence -- "Build with `xmtp-js`" -- is a much stronger start.

But it somewhat useful context, and a good lead-in to the security audit point. Maybe the "compromise" would be to move it right before "To learn more..."

Build with `xmtp-js` to provide messaging between blockchain wallet addresses, delivering on use cases such as wallet-to-wallet messaging and dapp-to-wallet notifications.

For a demonstration of the core concepts and capabilities of the `xmtp-js` client SDK, see the [example React app](https://github.com/xmtp/example-chat-react).

To learn more about XMTP and get answers to frequently asked questions, see <https://docs.xmtp.org/>.

## 🏗 **Breaking revisions**

Because `xmtp-js` is in active development, you should expect breaking revisions that might require you to adopt the latest SDK release to enable your app to continue working as expected.

> XMTP communicates about breaking revisions in the XMTP Discord community ([request access](https://xmtp.typeform.com/to/yojTJarb?utm_source=docs_home)), providing as much advance notice as possible. Additionally, breaking revisions in an `xmtp-js` release are described on the [Releases page](https://github.com/xmtp/xmtp-js/releases).

Author: Surfaced the Releases pages here, where we describe breaking changes in a release. 

Once we have a Changelog in the documentation, we can point there. Does this sound okay? =)

![x-red-sm](https://user-images.githubusercontent.com/510695/163488403-1fb37e86-c673-4b48-954e-8460ae4d4b05.png)

> **_Pre-stable_ XMTP client SDK for JavaScript applications**

Reviewer: I think we're ready to move on from "pre-stable"

> `xmtp-js` provides a TypeScript implementation of the XMTP client protocol for use with JavaScript applications, including React. It does not yet support React Native.

Reviewer: ```suggestion
`xmtp-js` provides a TypeScript implementation of the XMTP client protocol for use with JavaScript and React applications.
```

Build with `xmtp-js` to provide messaging between blockchain wallet addresses, delivering on use cases such as wallet-to-wallet messaging and dapp-to-wallet notifications.

For a demonstration of the core concepts and capabilities of the `xmtp-js` client SDK, see the [example React app](https://github.com/xmtp/example-chat-react).

To learn more about XMTP and get answers to frequently asked questions, see <https://docs.xmtp.org/>.

> ## 🏗 **XMTP-JS is in active development**

Reviewer: Maybe

```suggestion
## 🏗 Breaking revisions
```

since "active development" is mentioned a few words later

and then include a link to the changelog where we highlight breaking changes, like this: https://wagmi.sh/docs/migrating-to-03

> ![Security](https://img.shields.io/badge/security-unaudited-orange) ![Stability](https://img.shields.io/badge/code%20stability-low-orange)

Reviewer: these shields are probably too scary (visually), but we still need mention somewhere that the code is still unaudited

> This **pre-stable** release is publicly available for evaluation, feedback, and community contribution.

Reviewer: this sentence leaves out "building production apps", which is what we're trying to encourage now (also has "pre-stable")

Because XMTP-JS is in active development, you should expect breaking revisions that might require you to adopt the latest SDK version to enable your app to continue working as expected.

XMTP communicates about breaking revisions in the XMTP Discord community ([request access](https://xmtp.typeform.com/to/yojTJarb?utm_source=docs_home)), providing as much advance notice as possible.

> Feel free to contribute issues and PRs in this repo. The core team has limited bandwidth and may need a few days to review.

Reviewer: I think we can maybe replace this with a link to CONTRIBUTING.md. Don't think we should be saying "the core team has limited bandwidth" in the top-level README anymore.

## XMTP `dev` and `production` network environments

XMTP provides both `dev` and `production` network environments to support the development phases of your project.

Here are important points to understand when working with these network environments:

> - **DO NOT** share sensitive information on the `dev` network.

Reviewer: This reads as if we lack confidence in message encryption, which is the same in both environments. I am comfortable removing it.

> - XMTP does not regularly delete messages and keys from the `dev` and `production` networks. However, because XMTP is still in active development, we might need to delete messages and keys occasionally to maintain network stability. The likelihood of deletions might be greater on `dev` than on `production`. XMTP communicates about deletions in the XMTP Discord community ([request access](https://xmtp.typeform.com/to/yojTJarb?utm_source=docs_home)), providing as much advance notice as possible.

Reviewer: To me this still reads as a disclaimer that we might delete your messages from `production`. A starker contrast between the two networks might be something along the lines of.

> The `production` network is configured to store messages indefinitely. XMTP may occasionally delete messages and keys from the `dev` network, and will provide advance notice in Discord.
